src/Utilities/VulnerabilityRoutes.js

```diff
@@ -42,18 +43,28 @@ const Reports = lazy(() =>
     import(/* webpackChunkName: "Reports" */ '../Components/SmartComponents/Reports/ReportsPage')
 );
 
+export const AccountStatContext = createContext({
+    hasConventionalSystems: true,
+    hasEdgeDevices: false
+});
+
 const InsightsElement = ({ element: Element, title, globalFilterEnabled, ...elementProps }) => {
     let location = useLocation();
-    const [hasSystems, setHasSystems] = useState(true);
+    const getEdgeDevice = useGetSingleDevice();
```

Comment:
It seems almost the only use of useGetSingleDevice hook is to learn the total number of edge systems available for account. Why don't we create an async function that would make the same request but return the total_items value? So that we don't repeat https://github.com/RedHatInsights/vulnerability-ui/pull/2022/files#diff-bcc5cb503de3c25c192987043acea6e3dde8cd8166da6e5033aa045b77b6d731R61

Reply:
Nice idea. improved the useGetSingleDevice. However, I am afraid we can not convert it to plain function as there is a hook used inside.

Reply:
I believe there are many ways how to avoid using this hook. You can just use the instance from FEC and the inventory client (see more in Inventory src code):

```
import instance from '@redhat-cloud-services/frontend-components-utilities/interceptors';
export const hosts = new HostsApi(undefined, INVENTORY_API_BASE, instance);
```

Reply:
Not sure what is the best/most advised method. But I would be slightly against using hooks in simple functions that make API calls.

Reply:
Hmm, do we want to deal with maintaining the inventory-client version just for this request?

Reply:
It is maintained and updated by the platform team. I am not sure there are any disadvantages. Still many other apps use these clients, and there is nothing suggesting that we will deprecate it yet.

Reply:
The biggest disadvantage I think is to update the j-clients version when the backend makes a new release manually. This is not done by the platform. 

But, yeah having it would not have a sever impact neither. I will update the PR to use j-clients.

Reply:
@gkarat I made changes to use js-clients. Please take a look. Also, thank you for your input to improve the work.